### ISSUE TITLE
Documentation Tasks for 3.x

### ISSUE THREAD

**Issue by kanak** [monix/monix.io#29]
I hope this is OK, but I wanted to capture task ideas from @alexandru and @Avasil in https://github.com/monix/monix.io/pull/28#issuecomment-374912259 in a more visible  place and also add a checklist.

- [ ] Observable 
- [ ] Iterant (lots of documentation in Scaladocs)
- [ ] new Task features: bracket, TaskLocal, Fiber, Backoff
- [ ] TracingScheduler

**Comment by Avasil** [comment]
Yeah I think it's a good idea to list everything, then do separate issues etc. It should be easier to cooperate on it. Ideally I'd see table of contents to fill. :D

Other stuff off the top of my head is cats-effect integration (related to `Task` and `Iterant`) with example (I was thinking about showing how to interoperate between `Task` and `IO` received from somewhere like http4s)  

A lot of the changes can be noticed going through releases: https://monix.io/blog/2018/03/19/monix-v3.0.0-RC1.html

**Comment by alexandru** [comment]
👍 

FYI our competition for `Iterant` has a documentation microsite now, so checking that out is useful for inspiration 😊

https://functional-streams-for-scala.github.io/fs2/

This should also give us ideas about what's missing that's important.

Also the `Task` document needs a bigger section that talks of cancelation, like I wrote for `cats.effect.IO`: https://typelevel.org/cats-effect/datatypes/io.html#concurrency-and-cancelation

Just one piece of advice in case you start contributing — for documentation it's better to do it piecemeal, so prefer small PRs that can be merged faster.

Incomplete documentation is better than zero documentation 🙂

**Comment by alexandru** [comment]
Btw guys, helping with the docs is extremely valuable, so thanks for taking initiative ❤️

**Comment by alexandru** [comment]
FYI, I started to work on the `Iterant` document.

**Comment by Avasil** [comment]
Cool, I hope to have a bit more free time after a week or two and continue `Observable`

**Comment by kszlim** [comment]
`startAndForget` would be great to document too! Didn't realize monix had something like that until I really dug through the issues/code.

**Comment by Avasil** [comment]
@kszlim I have it on my todo list so I'll get to it eventually (though I focused on `Observable` more) unless you want to take a shot at it
